### PR TITLE
HSEARCH-3075 Release WorkPlan resources when a transaction is rolled back

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-3075

### CODE REVIEW

File: documentation/src/main/asciidoc/mapper-orm-indexing-automatic.asciidoc [hunk 1]
```diff
@@ -37,7 +37,8 @@ This is because queries are executed on the database side,
 One workaround is to <<mapper-orm-indexing-explicit,explicitly reindex>> after you run such queries.
 Indexing happens after transactions are committed::
 Indexes are not updated immediately, but only after transactions are successfully committed.
-That way, if a transaction is rolled back, the indexes will be left in a state consistent with the database.
+That way, if a transaction is rolled back, the indexes will be left in a state consistent with the database,
+discarding all the relative changes and releasing the relative resources.
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/mapper-orm-indexing-automatic.asciidoc]: I'm not sure "relative" can be used this way in this context.

Maybe: `discarding all the index changes that were planned during the transaction.`?

File: integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/work/IndexDocumentWorkExecutorIT.java [hunk 2]
```diff
@@ -79,6 +81,23 @@ public void checkAllDocumentsAreSearchable() {
 		Assertions.assertThat( query.fetchTotalHitCount() ).isEqualTo( NUMBER_OF_BOOKS );
 	}
 
+	@Test
+	public void discard() {
```
[the remark this answers is not in the document]
Comment (fax4ever) [reply integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/work/IndexDocumentWorkExecutorIT.java]: You're right. That's the wrong class.

File: documentation/src/main/asciidoc/mapper-orm-indexing-automatic.asciidoc (hunk 1 above)
Comment (fax4ever) [reply documentation/src/main/asciidoc/mapper-orm-indexing-automatic.asciidoc]: I translated literally from the Italian.. :P. Maybe `something that is relative to something else` doesn't have the same meaning in English... So let's change it.

**Comment by fax4ever** [comment]
Applied both that changes here.Thanks for the review.

File: integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/work/IndexDocumentWorkExecutorIT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (yrodiere) [reply integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/work/IndexDocumentWorkExecutorIT.java]: You didn't address this?

[the remark this answers is not in the document]
Comment (yrodiere) [reply integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/work/IndexDocumentWorkExecutorIT.java]: Ah, looks like you did, but you forgot to remove the method from `IndexDocumentWorkExecutorIT.java`.

[the remark this answers is not in the document]
Comment (fax4ever) [reply integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/work/IndexDocumentWorkExecutorIT.java]: You're right. Removed just now...

**Comment by yrodiere** [comment]
Merged, thanks!

**Comment by fax4ever** [comment]
Thank you!

**Pull request merged**